### DISCUSSION TITLE
Plans to Support MCAP File Format for Recording?

### DISCUSSION THREAD

**Discussion by bsirang** [eclipse-ecal/ecal#984] [Ideas]
Does the eCAL team have existing plans to support Foxglove's MCAP file format for recording (https://github.com/foxglove/mcap) instead of or alongside the existing hdf5 recording? I would love to see it. We want to move towards that format. Thanks!

**Comment by KerstinKeller** [comment]
Hi, yes actually we do have plans to support it!
We did a few benchmarks and write performance for synthetic data seems to outperform the native hdf5 write performance.
It will be coming later this year as an additional format, maybe even for 5.12.

We also have a conversion tool to convert hdf5 to mcap, for usage in Foxglove Studio, as well as a live eCAL data connection over Foxglove websocket protocol, which we also have in the pipeline to opensource.

**Reply by bsirang** [reply]
Great news! What are the chances it will come in v5.12? It would be great if it made it in the next release.

**Reply by chengguizi** [reply]
+1 for support of more format common in the ecosystem, including MCAP!

**Reply by KerstinKeller** [reply]
We can prioritize it a bit, if you would like to see it happen.
Technically, it is not very challenging, however there are a few questionmarks still:
- would you a as a user like to set settings such as compression format / ratio / ... or should eCAL player set / adapt these based on write performance. 

So maybe we can include the format as an experimental feature for 5.12, but we will allow ourselves to make some breaking changes for the upcoming releases.
